include/ua_plugin_securitypolicy.h

```diff
@@ -0,0 +1,424 @@
+                                         UA_ByteString *out);
+
+    /**
+     * \brief Calculates the padding size for a message with the specified amount of bytes.
+     *
+     * \param securityPolicy the securityPolicy the function is invoked on.
+     * \param bytesToWrite the size of the payload plus the sequence header, since both need to be encoded
+     * \param paddingSize out parameter. Will contain the paddingSize byte.
+     * \param extraPaddingSize out parameter. Will contain the extraPaddingSize. If no extra padding is needed, this is 0.
+     * \return the total padding size consisting of high and low bytes.
+     */
+    UA_UInt16 (*const calculatePadding)(const UA_SecurityPolicy *securityPolicy,
+                                        size_t bytesToWrite,
+                                        UA_Byte *paddingSize,
+                                        UA_Byte *extraPaddingSize);
+
+    const UA_SecurityPolicySigningModule signingModule;
+
+    const size_t signingKeyLength;
+    const size_t encryptingKeyLength;
+    const size_t encryptingBlockSize;
+} UA_SecurityPolicySymmetricModule;
+
+struct UA_Endpoint_SecurityContext {
+    UA_StatusCode (*const init)(const UA_SecurityPolicy *securityPolicy,
+                                const void *initData,
+                                void **pp_contextData);
+
+    UA_StatusCode (*const deleteMembers)(const UA_SecurityPolicy *securityPolicy,
+                                         void *endpointContext);
+
+    UA_StatusCode (*const setLocalPrivateKey)(const UA_SecurityPolicy *securityPolicy,
+                                              const UA_ByteString *privateKey,
+                                              void *endpointContext);
+
+    UA_StatusCode (*const setServerCertificate)(const UA_SecurityPolicy *securityPolicy,
+                                                const UA_ByteString *certificate,
+                                                void *endpointContext);
+
+    const UA_ByteString *(*const getServerCertificate)(const UA_SecurityPolicy *securityPolicy,
+                                                       const void *endpointContext);
+
+    UA_StatusCode (*const setCertificateTrustList)(const UA_SecurityPolicy *securityPolicy,
+                                                   const UA_ByteString *trustList,
+                                                   void *endpointContext);
+
+    UA_StatusCode (*const setCertificateRevocationList)(const UA_SecurityPolicy *securityPolicy,
+                                                        const UA_ByteString *revocationList,
+                                                        void *endpointContext);
+
+    /**
+    * Gets the signature size that depends on the local private key.
+    * This will return 0 as long as no remote certificate was set.
+    */
+    size_t (*const getLocalAsymSignatureSize)(const UA_SecurityPolicy *securityPolicy,

```

Comment:
My thought behind this was, that the size is stored in the endpoint context, and since we want to retrieve it from there, it's part of the endpoint context. If we were to make this part of the signing module, it would exist for both symmetric and asymmetric module. Maybe it's better that way, since we can then remove the attribute and just have the function return the size in both cases. That would be a bit more consistent i guess.